app/services/UserServiceImpl.js

```diff
@@ -29,7 +29,28 @@ module.exports = function(userDao) {
     });
   };
 
+  const findByUsername = function(username, password) {
+    return new Promise((resolve, reject) => {
+      _userDao.findByUsername(username)
+        .then((user) => {
+          return isMatchingPassword(password, user.password)
```

Comment:
And yes I'm getting confused with the other PR lol. Think I'll have a merge conflict but should be able to get it figured out.